platform/core/src/redux/reducers/viewports.js

```diff
@@ -25,6 +25,40 @@ const DEFAULT_STATE = {
   viewportSpecificData: {},
 };
 
+/**
+ *  Take the new number of rows and columns, delete all not used viewport data and also set
+ *  active viewport as default in case current one is not available anymore.
+ *
+ * @param {Object} action -
+ * @param {Number} action.numRows -
+ * @param {Number} action.numColumns -
+ * @param {Object} currentViewportSpecificData -
+ * @param {Number} currentActiveViewportIndex -
+ * @returns
+ */
+const handleViewportDeletions = (action, currentViewportSpecificData, currentActiveViewportIndex) => {
```

Comment:
This is to make sure, whenever we reduce the number of viewports, we only keep the specific data for the ones that are still visible/enabled. 

Same for the activeViewportIndex, to make sure that the active viewport index exists